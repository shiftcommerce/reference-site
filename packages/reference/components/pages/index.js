import CategorySidebar from './categorySidebar'

const PageTemplates = {
  category_sidebar_template: CategorySidebar
}

export default PageTemplates
