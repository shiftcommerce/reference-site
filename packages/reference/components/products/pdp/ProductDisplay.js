// Libraries
import { Component } from 'react'
import classNames from 'classnames'
import Link from 'next/link'

// Components
import Carousel from './Carousel'
import ProductNavBar from '../../navigation/ProductNavBar'
import ProductPrice from './ProductPrice'

// Objects
import Button from '../../../objects/Button'
import SizeSelector from '../../../objects/SizeSelector'

// Fixtures
import Images from '../../../spec/fixtures/dummyPDPimages.fixture'

class ProductDisplay extends Component {
  renderRatingStars (product) {
    const rating = (product.rating || 0)

    return (
      [1, 2, 3, 4, 5].map((key) =>
        <span key={key} className={classNames({ 'c-product-display__info-rating--fill': key <= rating, 'c-product-display__info-rating--empty': key > rating })}>
          &#9733;
        </span>
      )
    )
  }

  renderCarousel () {
    return <>
      <div className='c-product-display__gallery'>
        <div className='c-carousel'>
          <Carousel className='c-carousel-slider' product={this.props.product} />
        </div>
      </div>
    </>
  }

  renderButtons () {
    const { product, addToBag } = this.props

    return <>
      <div className='c-product-display__buttons'>
        <div className='c-product-display__buttons-section1'>
          <div className='c-product-display__buttons-title'>
            {product.title}
          </div>
          <div className='c-product-display__buttons-price'>
            <ProductPrice variants={product.variants} />
          </div>
        </div>
        <div className='c-product-display__buttons-section2'>
          <div className='c-product-display__buttons-basket'>
            <Button className='c-product-display__buttons-basket-icon' label='ADD TO BASKET' status='primary' size='lrg' aria-label='Add to Basket' onClick={addToBag} />
          </div>
          <div className='c-product-display__buttons-buy'>
            <Button className='c-product-display__buttons-buy-icon' label='BUY' status='secondary' size='lrg' aria-label='Buy' />
          </div>
        </div>
      </div>
    </>
  }

  renderDescription () {
    const { product } = this.props

    return <>
      <div className='c-product-display__description'>
        <h1 className='c-product-display__description-title'>Product Details</h1>
        <label htmlFor='description' className='c-product-dispay__label' />
        <input type='checkbox' id='description' />
        <span className='c-product-display__arrow' />
        <div className='c-product-display__description-text' dangerouslySetInnerHTML={{ __html: product.description }} />
      </div>
    </>
  }

  renderInfo () {
    const { product, changeSize, sku } = this.props
    const productColour = product.meta_attributes.master_colour.value

    return <>
      <div className='c-product-display__info'>
        <div className='c-product-display__info-title'>
          {product.title}
        </div>
        <div className='c-product-display__info-price'>
          <ProductPrice variants={product.variants} />
        </div>
        <div className='c-product-display__info-rating'>
          { this.renderRatingStars(product) }
        </div>
        <p className='c-product-display__info-reviews'>Read Reviews</p>
        <div className='c-product-display__info-colour'>
          <h1 className='c-product-display__info-colour-title'>Colour</h1>
          <input type='radio' id='colour' style={{ backgroundColor: productColour }} />
          <label htmlFor='colour'>{productColour}</label>
        </div>
        <p className='c-product-display__info-size-chart'>Find your recommended size</p>
        <div className='c-product-display__info-size'>
          <SizeSelector onChange={changeSize} value={sku} name='line_item[item_id]' prompt='Please Select' label='Size' variants={product.variants} aria-label='Size Selector' />
        </div>
      </div>
    </>
  }

  renderFit (meta) {
    if (meta.silhouette) {
      return <>
        <p className='c-product-display__size-text'>- {meta.silhouette.value}</p>
      </>
    }
  }

  renderFeature (meta) {
    if (meta.product_feature) {
      return (
        <p className='c-product-display__size-text'>- {meta.product_feature.value}</p>
      )
    }
  }

  renderSizeAndFit () {
    const { product: { meta_attributes } } = this.props

    if (meta_attributes.product_feature || meta_attributes.silhouette) {
      return <>
        <div className='c-product-display__size'>
          <h1 className='c-product-display__size-title'>Size & Fit</h1>
          <label htmlFor='size' className='c-product-dispay__label' />
          <input type='checkbox' id='size' />
          <span className='c-product-display__arrow' />
          { this.renderFeature(meta_attributes) }
          { this.renderFit(meta_attributes) }
        </div>
      </>
    }
  }

  renderDetailsAndCare () {
    const { product } = this.props
    const cleaning = product.meta_attributes.care_instructions.value
    const fabric = product.meta_attributes.fabric.value

    return <>
      <div className='c-product-display__care'>
        <h1 className='c-product-display__care-title'>Details & Care</h1>
        <label htmlFor='care' className='c-product-dispay__label' />
        <input type='checkbox' id='care' />
        <span className='c-product-display__arrow' />
        <p className='c-product-display__care-text'>- {fabric}</p>
        <p className='c-product-display__care-text'>- {cleaning}</p>
      </div>
    </>
  }

  // Currently using dummy data from fixture
  renderRelatedProducts () {
    return <>
      <div className='c-product-display__related'>
        <h1 className='c-product-display__related-title'><span>you may also like</span></h1>
        <div className='c-product-display__related-images'>
          { Images.asset_files.map((image, index) => {
            return (
              <Link href={image.link} key={index}>
                <div className='c-product-display__related-images-link'>
                  <img className='c-product-display__related-images-file' src={image.asset_file} />
                  <p className='c-product-display__related-images-text'>{image.text}</p>
                  <p className='c-product-display__related-images-price'>{image.price}</p>
                </div>
              </Link>
            )
          })
        }
        </div>
      </div>
    </>
  }

  render () {
    return <>
      <div className='c-product-display'>
        <ProductNavBar />
        <div className='c-product-display__body'>
          { this.renderCarousel() }
          { this.renderButtons() }
          { this.renderInfo() }
          { this.renderDescription() }
          { this.renderSizeAndFit() }
          { this.renderDetailsAndCare() }
          { this.renderRelatedProducts() }
        </div>
      </div>
    </>
  }
}

export default ProductDisplay
