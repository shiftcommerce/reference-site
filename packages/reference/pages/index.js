import { Component } from 'react'

// components
import Layout from '../components/Layout'

class Home extends Component {
  render () {
    return <Layout>
      <div>Welcome To Shift Commerce Front End React</div>
    </Layout>
  }
}

export default Home
