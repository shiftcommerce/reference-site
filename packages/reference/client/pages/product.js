// Libraries
import { Component } from 'react'
import { connect } from 'react-redux'

// Lib
import renderComponents from '../lib/render-components'

// Components
import Loading from '../components/loading'
import ProductDisplay from '../components/products/display/product-display'

// Actions
import { addToCart } from '../actions/cart-actions'
import { readProduct } from '../actions/product-actions'

export class Product extends Component {
  constructor (props) {
    super(props)
    // TODO: Update to use redux
    this.state = {
      sku: '',
      variant: '',
      variantId: null,
      quantity: 1,
      price: '',
      stockAvailableLevel: 0
    }

    this.changeVariant = this.changeVariant.bind(this)
    this.addToBag = this.addToBag.bind(this)
  }

  static async getInitialProps ({ reduxStore, req, query }) {
    const { id } = query
    const isServer = !!req

    if (isServer) {
      await reduxStore.dispatch(readProduct(id))
    }

    return { id: id }
  }

  componentDidMount () {
    const { dispatch, id } = this.props
    dispatch(readProduct(id))
  }

  addToBag () {
    const { quantity, sku, variant, stockAvailableLevel, price } = this.state
    const { product } = this.props

    if (quantity !== '' && sku !== '') {
      const lineItem = {
        sku: sku,
        title: product.title,
        variant: variant,
        quantity: parseInt(quantity),
        stockAvailableLevel: stockAvailableLevel,
        price: price,
        imageUrl: (product.asset_files[0] && product.asset_files[0].s3_url),
        productSku: product.sku,
        productID: product.id,
        slug: product.slug,
        canonicalPath: product.canonical_path,
        bundles: product.bundles
      }
      this.props.dispatch(addToCart(lineItem))
    } else {
      /* eslint-disable */
      alert('Please select a size')
      /* eslint-enable */
    }
  }

  changeVariant (e) {
    this.setState({
      stockAvailableLevel: parseInt(e.target.options[e.target.selectedIndex].dataset.stockAvailableLevel, 10),
      sku: e.target.value,
      variant: e.target.options[e.target.selectedIndex].text,
      variantId: e.target.options[e.target.selectedIndex].dataset.variantId,
      price: e.target.options[e.target.selectedIndex].dataset.price
    })
  }

  render () {
    const { product, product: { loading, error, template } } = this.props

    if (loading) {
      return (
        <Loading />
      )
    } else if (error || Object.keys(product).length === 0) {
      return (
        <h1>Unable to load product.</h1>
      )
    } else {
      const templateSection = template && template.sections && template.sections.slice(-1).pop()
      const components = templateSection && templateSection.components
      const selectedVariant = product.variants.find(variant => variant.id === this.state.variantId)

      return (
        <>
          <ProductDisplay
            product={product}
            changeVariant={this.changeVariant}
            addToBag={this.addToBag}
            clickToBuy={this.clickToBuy}
            selectedVariant={selectedVariant}
          />
          { components && renderComponents(components) }
        </>
      )
    }
  }
}

function mapStateToProps (state) {
  const { menu, product, cart } = state
  return { menu, product, cart }
}

export default connect(mapStateToProps)(Product)
