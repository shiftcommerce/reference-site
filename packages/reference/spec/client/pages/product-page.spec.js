// Libraries
import { Provider } from 'react-redux'
import { createMockStore } from 'redux-test-utils'

// Page
import { Product } from './../../../client/pages/product'

// Components
import Loading from './../../../client/components/loading'
import ProductDisplay from './../../../client/components/products/display/product-display'

// Fixtures
import product from './../../fixtures/product'

describe('Product page', () => {
  describe('loading and error states', () => {
    test('displays loading spinner, when there is no error and product is loading', () => {
      // Arrange
      const product = {
        loading: true,
        error: false,
        data: []
      }

      const dispatch = jest.fn()

      // Act
      const wrapper = mount(
        <Loading product={product} dispatch={dispatch} />
      )

      // Assert
      expect(wrapper).toMatchSnapshot()
      expect(wrapper.find('.c-loading')).toHaveClassName('c-loading')
    })

    test('displays an error message when there is an error loading the product', () => {
      // Arrange
      const product = {
        loading: false,
        error: true,
        data: []
      }

      const dispatch = jest.fn()

      // Act
      const wrapper = mount(
        <Provider store={createMockStore()}>
          <Product product={product} dispatch={dispatch} />
        </Provider>
      )

      // Assert
      expect(wrapper).toMatchSnapshot()
      expect(wrapper).toIncludeText('Unable to load product.')
    })

    test('displays relevant message, when product data is empty', () => {
      // Arrange
      const dispatch = jest.fn()

      // Act
      const wrapper = mount(
        <Provider store={createMockStore()}>
          <Product product={{}} dispatch={dispatch} />
        </Provider>
      )

      // Assert
      expect(wrapper).toMatchSnapshot()
      expect(wrapper).toIncludeText('Unable to load product.')
    })
  })

  test('renders a product once data is present', () => {
    // Arrange
    const dispatch = jest.fn()

    // Act
    const wrapper = mount(
      <Provider store={createMockStore()}>
        <Product product={product} dispatch={dispatch} />
      </Provider>
    )

    // Assert
    expect(wrapper).toMatchSnapshot()

    expect(wrapper).toIncludeText(product.title)
    expect(wrapper).toIncludeText(product.min_current_price)

    expect(wrapper.find(ProductDisplay)).toExist()
  })
})
