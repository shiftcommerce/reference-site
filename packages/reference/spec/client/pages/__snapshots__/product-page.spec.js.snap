// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product page loading and error states displays an error message when there is an error loading the product 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <Product
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    product={
      Object {
        "data": Array [],
        "error": true,
        "loading": false,
      }
    }
  >
    <h1>
      Unable to load product.
    </h1>
  </Product>
</Provider>
`;

exports[`Product page loading and error states displays loading spinner, when there is no error and product is loading 1`] = `
<Loading
  dispatch={[MockFunction]}
  product={
    Object {
      "data": Array [],
      "error": false,
      "loading": true,
    }
  }
>
  <Loading
    className="c-loading"
    color="#431048"
    delay={0}
    height={50}
    type="spin"
    width={50}
  >
    <div
      className="c-loading"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 32 32\\">
  <path opacity=\\".25\\" d=\\"M16 0 A16 16 0 0 0 16 32 A16 16 0 0 0 16 0 M16 4 A12 12 0 0 1 16 28 A12 12 0 0 1 16 4\\"/>
  <path d=\\"M16 0 A16 16 0 0 1 32 16 L28 16 A12 12 0 0 0 16 4z\\">
    <animateTransform attributeName=\\"transform\\" type=\\"rotate\\" from=\\"0 16 16\\" to=\\"360 16 16\\" dur=\\"0.8s\\" repeatCount=\\"indefinite\\" />
  </path>
</svg>
",
        }
      }
      style={
        Object {
          "fill": "#431048",
          "height": 50,
          "width": 50,
        }
      }
    />
  </Loading>
</Loading>
`;

exports[`Product page loading and error states displays relevant message, when product data is empty 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <Product
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    product={Object {}}
  >
    <h1>
      Unable to load product.
    </h1>
  </Product>
</Provider>
`;

exports[`Product page renders a product once data is present 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getAction": [Function],
      "getActions": [Function],
      "getState": [Function],
      "isActionDispatched": [Function],
      "isActionTypeDispatched": [Function],
      "subscribe": [Function],
    }
  }
>
  <Product
    dispatch={
      [MockFunction] {
        "calls": Array [
          Array [
            [Function],
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
    product={
      Object {
        "asset_files": Array [
          Object {
            "asset_file": "1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
            "canonical_path": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
            "created_at": "2018-09-24T15:41:57Z",
            "deleted_at": null,
            "file_content_content_type": "image/jpeg",
            "file_content_filename": "SEPIA_nonslip_tray_420_210mm.jpg",
            "file_content_size": 24096,
            "id": "139",
            "image_height": 600,
            "image_width": 600,
            "meta_attributes": Object {},
            "name": "SEPIA nonslip tray 420×210mm.jpg",
            "reference": "1d6cc081-cbe6-4e90-b2e0-6f404ed68e64",
            "s3_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
            "updated_at": "2018-09-24T15:43:05Z",
          },
        ],
        "attribute_merchandising": Object {},
        "available_to_browse": false,
        "average_rating": 0,
        "bundles": Array [],
        "canonical_path": "/products/seed_product_51",
        "category_ids_cache": Array [
          58,
          59,
        ],
        "content_type": "markdown",
        "created_at": "2018-09-24T15:43:05Z",
        "deleted_at": null,
        "description": "Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
        "error": false,
        "ewis_status": "not_ewis_eligible",
        "has_assets": true,
        "id": "175",
        "loading": false,
        "max_current_price": 97.68,
        "max_price": 97.68,
        "merchandised": true,
        "merged_attribute_merchandising": Object {},
        "meta_attributes": Object {},
        "min_current_price": 16.95,
        "min_price": 16.95,
        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
        "published": true,
        "rating": 0,
        "reference": "seed_product_51",
        "sellable": true,
        "seo_meta_description": "Product 51 - Ut quam iusto ea.",
        "seo_meta_keywords": "in, voluptatem, aut, fugit, quo",
        "seo_meta_title": "SEPIA nonslip tray 420×210mm",
        "slug": "seed_product_51",
        "template": null,
        "title": "SEPIA nonslip tray 420×210mm",
        "total_ratings": null,
        "updated_at": "2018-11-05T11:23:18Z",
        "variants": Array [
          Object {
            "available_to_browse": false,
            "available_to_order": false,
            "created_at": "2018-09-24T15:43:05Z",
            "current_price": 16.95,
            "deleted_at": null,
            "description": "Product 51 - variant 1

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
            "ewis_eligible": false,
            "expected_availability_date": null,
            "has_assets": true,
            "id": "343",
            "meta_attributes": Object {},
            "ordered_total": 10,
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
            "position": null,
            "price": 16.95,
            "price_includes_taxes": true,
            "product_id": 175,
            "reference": "dolor_50_1",
            "sku": "9337040128595",
            "stock_allocated_level": 10,
            "stock_available_level": 0,
            "stock_level": 1,
            "tax_code": "vat",
            "title": "Variant 1 for SEPIA nonslip tray 420×210mm",
            "updated_at": "2018-11-05T11:23:16Z",
          },
          Object {
            "available_to_browse": false,
            "available_to_order": false,
            "created_at": "2018-09-24T15:43:05Z",
            "current_price": 97.68,
            "deleted_at": null,
            "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
            "ewis_eligible": false,
            "expected_availability_date": null,
            "has_assets": true,
            "id": "344",
            "meta_attributes": Object {},
            "ordered_total": 1,
            "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
            "position": null,
            "price": 97.68,
            "price_includes_taxes": true,
            "product_id": 175,
            "reference": "voluptas_50_2",
            "sku": "0927967530244",
            "stock_allocated_level": 1,
            "stock_available_level": 0,
            "stock_level": 1,
            "tax_code": "vat",
            "title": "Variant 2 for SEPIA nonslip tray 420×210mm",
            "updated_at": "2018-10-31T10:24:00Z",
          },
        ],
        "web_product_description": "Product 51 - Perspiciatis recusandae voluptatem culpa.",
        "web_product_name": "SEPIA nonslip tray 420×210mm",
      }
    }
  >
    <ProductDisplay
      addToBag={[Function]}
      changeVariant={[Function]}
      product={
        Object {
          "asset_files": Array [
            Object {
              "asset_file": "1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
              "canonical_path": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
              "created_at": "2018-09-24T15:41:57Z",
              "deleted_at": null,
              "file_content_content_type": "image/jpeg",
              "file_content_filename": "SEPIA_nonslip_tray_420_210mm.jpg",
              "file_content_size": 24096,
              "id": "139",
              "image_height": 600,
              "image_width": 600,
              "meta_attributes": Object {},
              "name": "SEPIA nonslip tray 420×210mm.jpg",
              "reference": "1d6cc081-cbe6-4e90-b2e0-6f404ed68e64",
              "s3_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
              "updated_at": "2018-09-24T15:43:05Z",
            },
          ],
          "attribute_merchandising": Object {},
          "available_to_browse": false,
          "average_rating": 0,
          "bundles": Array [],
          "canonical_path": "/products/seed_product_51",
          "category_ids_cache": Array [
            58,
            59,
          ],
          "content_type": "markdown",
          "created_at": "2018-09-24T15:43:05Z",
          "deleted_at": null,
          "description": "Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
          "error": false,
          "ewis_status": "not_ewis_eligible",
          "has_assets": true,
          "id": "175",
          "loading": false,
          "max_current_price": 97.68,
          "max_price": 97.68,
          "merchandised": true,
          "merged_attribute_merchandising": Object {},
          "meta_attributes": Object {},
          "min_current_price": 16.95,
          "min_price": 16.95,
          "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
          "published": true,
          "rating": 0,
          "reference": "seed_product_51",
          "sellable": true,
          "seo_meta_description": "Product 51 - Ut quam iusto ea.",
          "seo_meta_keywords": "in, voluptatem, aut, fugit, quo",
          "seo_meta_title": "SEPIA nonslip tray 420×210mm",
          "slug": "seed_product_51",
          "template": null,
          "title": "SEPIA nonslip tray 420×210mm",
          "total_ratings": null,
          "updated_at": "2018-11-05T11:23:18Z",
          "variants": Array [
            Object {
              "available_to_browse": false,
              "available_to_order": false,
              "created_at": "2018-09-24T15:43:05Z",
              "current_price": 16.95,
              "deleted_at": null,
              "description": "Product 51 - variant 1

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
              "ewis_eligible": false,
              "expected_availability_date": null,
              "has_assets": true,
              "id": "343",
              "meta_attributes": Object {},
              "ordered_total": 10,
              "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
              "position": null,
              "price": 16.95,
              "price_includes_taxes": true,
              "product_id": 175,
              "reference": "dolor_50_1",
              "sku": "9337040128595",
              "stock_allocated_level": 10,
              "stock_available_level": 0,
              "stock_level": 1,
              "tax_code": "vat",
              "title": "Variant 1 for SEPIA nonslip tray 420×210mm",
              "updated_at": "2018-11-05T11:23:16Z",
            },
            Object {
              "available_to_browse": false,
              "available_to_order": false,
              "created_at": "2018-09-24T15:43:05Z",
              "current_price": 97.68,
              "deleted_at": null,
              "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
              "ewis_eligible": false,
              "expected_availability_date": null,
              "has_assets": true,
              "id": "344",
              "meta_attributes": Object {},
              "ordered_total": 1,
              "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
              "position": null,
              "price": 97.68,
              "price_includes_taxes": true,
              "product_id": 175,
              "reference": "voluptas_50_2",
              "sku": "0927967530244",
              "stock_allocated_level": 1,
              "stock_available_level": 0,
              "stock_level": 1,
              "tax_code": "vat",
              "title": "Variant 2 for SEPIA nonslip tray 420×210mm",
              "updated_at": "2018-10-31T10:24:00Z",
            },
          ],
          "web_product_description": "Product 51 - Perspiciatis recusandae voluptatem culpa.",
          "web_product_name": "SEPIA nonslip tray 420×210mm",
        }
      }
    >
      <div
        className="c-product-display"
      >
        <div
          className="c-product-display__body"
        >
          <div
            className="c-product-display__content-image"
          >
            <div
              className="c-product-display__gallery"
            >
              <Image
                className="c-product-display__gallery-image"
                key="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
                src="https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg"
              >
                <div
                  className="c-product-display__gallery-image"
                >
                  <LazyLoad
                    debounce={true}
                    elementType="div"
                    offset={0}
                    offsetBottom={0}
                    offsetHorizontal={0}
                    offsetLeft={0}
                    offsetRight={0}
                    offsetTop={0}
                    offsetVertical={0}
                    throttle={250}
                  >
                    <div
                      className="LazyLoad"
                      style={
                        Object {
                          "height": undefined,
                          "width": undefined,
                        }
                      }
                    />
                  </LazyLoad>
                </div>
              </Image>
            </div>
          </div>
          <div
            className="c-product-display__content-details"
          >
            <div
              className="c-product-display__info"
            >
              <div
                className="c-product-display__info-headings"
              >
                <div
                  className="c-product-display__info-title"
                >
                  SEPIA nonslip tray 420×210mm
                </div>
              </div>
              <div
                className="c-product-display__info-price"
              >
                <ProductPrice
                  maxPrice={97.68}
                  minPrice={16.95}
                >
                  <div
                    className="c-product-price"
                  >
                    <span>
                      £
                      16.95
                       - £
                      97.68
                    </span>
                  </div>
                </ProductPrice>
              </div>
              <div
                className="c-product-display__info-rating"
              >
                <span
                  className="c-product-display__info-rating--empty"
                  key="1"
                >
                  ★
                </span>
                <span
                  className="c-product-display__info-rating--empty"
                  key="2"
                >
                  ★
                </span>
                <span
                  className="c-product-display__info-rating--empty"
                  key="3"
                >
                  ★
                </span>
                <span
                  className="c-product-display__info-rating--empty"
                  key="4"
                >
                  ★
                </span>
                <span
                  className="c-product-display__info-rating--empty"
                  key="5"
                >
                  ★
                </span>
                <p
                  className="c-product-display__info-reviews"
                >
                  Read Reviews
                </p>
              </div>
              <div
                className="c-product-display__info-variant"
              >
                <VariantSelector
                  aria-label="Variant Selector"
                  name="line_item[item_id]"
                  onChange={[Function]}
                  prompt="Select a Product"
                  variants={
                    Array [
                      Object {
                        "available_to_browse": false,
                        "available_to_order": false,
                        "created_at": "2018-09-24T15:43:05Z",
                        "current_price": 16.95,
                        "deleted_at": null,
                        "description": "Product 51 - variant 1

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
                        "ewis_eligible": false,
                        "expected_availability_date": null,
                        "has_assets": true,
                        "id": "343",
                        "meta_attributes": Object {},
                        "ordered_total": 10,
                        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
                        "position": null,
                        "price": 16.95,
                        "price_includes_taxes": true,
                        "product_id": 175,
                        "reference": "dolor_50_1",
                        "sku": "9337040128595",
                        "stock_allocated_level": 10,
                        "stock_available_level": 0,
                        "stock_level": 1,
                        "tax_code": "vat",
                        "title": "Variant 1 for SEPIA nonslip tray 420×210mm",
                        "updated_at": "2018-11-05T11:23:16Z",
                      },
                      Object {
                        "available_to_browse": false,
                        "available_to_order": false,
                        "created_at": "2018-09-24T15:43:05Z",
                        "current_price": 97.68,
                        "deleted_at": null,
                        "description": "Product 51 - variant 2

Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
                        "ewis_eligible": false,
                        "expected_availability_date": null,
                        "has_assets": true,
                        "id": "344",
                        "meta_attributes": Object {},
                        "ordered_total": 1,
                        "picture_url": "https://shift-platform-dev.s3-eu-west-1.amazonaws.com/uploads/asset_file/asset_file/139/1537803717.9168365-SEPIA_nonslip_tray_420_210mm.jpg",
                        "position": null,
                        "price": 97.68,
                        "price_includes_taxes": true,
                        "product_id": 175,
                        "reference": "voluptas_50_2",
                        "sku": "0927967530244",
                        "stock_allocated_level": 1,
                        "stock_available_level": 0,
                        "stock_level": 1,
                        "tax_code": "vat",
                        "title": "Variant 2 for SEPIA nonslip tray 420×210mm",
                        "updated_at": "2018-10-31T10:24:00Z",
                      },
                    ]
                  }
                >
                  <div
                    className="o-variant-selector"
                  >
                    <select
                      aria-label="Variant Selector"
                      name="line_item[item_id]"
                      onChange={[Function]}
                    >
                      <option
                        aria-posinset="0"
                        aria-setsize={2}
                        role="option"
                        value=""
                      >
                        Select a Product
                      </option>
                      <option
                        aria-posinset={1}
                        aria-setsize={2}
                        data-price={16.95}
                        data-stock-available-level={0}
                        data-variant-id="343"
                        key="0"
                        role="option"
                        value="9337040128595"
                      >
                        Variant 1 for SEPIA nonslip tray 420×210mm
                      </option>
                      <option
                        aria-posinset={2}
                        aria-setsize={2}
                        data-price={97.68}
                        data-stock-available-level={0}
                        data-variant-id="344"
                        key="1"
                        role="option"
                        value="0927967530244"
                      >
                        Variant 2 for SEPIA nonslip tray 420×210mm
                      </option>
                    </select>
                  </div>
                </VariantSelector>
              </div>
            </div>
            <div
              className="c-product-display__buttons"
            >
              <Button
                aria-label="Add to Basket"
                className="c-product-display__buttons-basket o-button--sml"
                label="ADD TO BASKET"
                onClick={[Function]}
                status="positive"
              >
                <div
                  className="o-button__container"
                >
                  <button
                    aria-label="Add to Basket"
                    className="c-product-display__buttons-basket o-button--sml o-button o-button--positive"
                    disabled={false}
                    onClick={[Function]}
                    role="button"
                    tabIndex="0"
                  >
                    ADD TO BASKET
                  </button>
                </div>
              </Button>
            </div>
            <div
              className="c-product-display__body-dropdowns"
            >
              <div
                className="c-product-display__description"
              >
                <h1
                  className="c-product-display__description-title"
                >
                  Product Details
                </h1>
                <label
                  className="c-product-dispay__label"
                  htmlFor="description"
                />
                <input
                  id="description"
                  type="checkbox"
                />
                <span
                  className="c-product-display__arrow"
                />
                <div
                  className="c-product-display__description-text"
                  dangerouslySetInnerHTML={
                    Object {
                      "__html": "Assam in north-eastern India grows exceptional tea in the lush soil of the Brahmaputra valley – and we pick ours at the peak of the season from some of the region's finest estates during the second flush harvest. It's rich, strong and full-bodied with malty notes.",
                    }
                  }
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </ProductDisplay>
  </Product>
</Provider>
`;
